package_bag/routines.py

```diff
@@ -192,6 +129,56 @@ def process_bag(self, bag):
         raise NotImplementedError("You must implement a `process_bag` method")
 
 
+class BagDiscoverer(BaseRoutine):
+    """
+    Validates bag structure and bag info file, renames bag with unique ID
+    """
+    start_process_status = Bag.DOWNLOADED
+    in_process_status = Bag.DISCOVERING
+    end_process_status = Bag.DISCOVERED
+    success_message = "Bag renamed, unpacked, and validated."
+    idle_message = "No bags were found."
+
+    def __init__(self):
+        self.src_dir = settings.SRC_DIR
+        self.tmp_dir = settings.TMP_DIR
+        for dir in [self.src_dir, self.tmp_dir]:
+            if not isdir(dir):
+                raise Exception("Directory does not exist", dir)
+
+    def process_bag(self, bag):
+        bag.bag_path = self.unpack_rename(bag)
+        bag.save()
```

Comment:
You're calling bag.save() twice here. You _may_ not need to call it at all, since that gets called in the BaseRoutine.